test/jdk/jdk/internal/util/ByteArray/ReadWriteValues.java

```diff
@@ -148,40 +148,192 @@ void testPutFloat() {
     void testGetDouble() {
         doubles().forEach(expected -> {
             RefImpl.putDouble(BUFF, OFFSET, expected);
-            double actual = BitsProxy.getDouble(BUFF, OFFSET);
+            double actual = ByteArray.getDouble(BUFF, OFFSET);
             assertEquals(expected, actual);
         });
     }
 
     @Test
     void testPutDouble() {
         doubles().forEach(expected -> {
-            BitsProxy.putDouble(BUFF, OFFSET, expected);
+            ByteArray.setDouble(BUFF, OFFSET, expected);
             double actual = RefImpl.getDouble(BUFF, OFFSET);
             assertEquals(expected, actual);
         });
     }
 
+    @Test
+    void testPutUnsignedShort() {
+        longs().forEach(l -> {
+            int expected = Short.toUnsignedInt((short) l);
+            ByteArray.setUnsignedShort(BUFF, OFFSET, expected);
+            int actual = Short.toUnsignedInt(RefImpl.getShort(BUFF, OFFSET));
+            assertEquals(expected, actual);
+        });
+    }
+
+
     // Unusual cases
 
     @Test
     void testNullArray() {
-        assertThrowsOriginal(NullPointerException.class, () -> BitsProxy.getInt(null, OFFSET));
-        assertThrowsOriginal(NullPointerException.class, () -> BitsProxy.putInt(null, OFFSET, 1));
+        assertThrowsOriginal(NullPointerException.class, () -> ByteArray.getInt(null, OFFSET));
+        assertThrowsOriginal(NullPointerException.class, () -> ByteArray.setInt(null, OFFSET, 1));
     }
 
     @Test
     void testNegArg() {
-        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> BitsProxy.getInt(BUFF, -1));
-        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> BitsProxy.putInt(BUFF, -1, 1));
+        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> ByteArray.getInt(BUFF, -1));
+        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> ByteArray.setInt(BUFF, -1, 1));
     }
 
     @Test
     void testOutOfBounds() {
-        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> BitsProxy.getInt(BUFF, BUFF.length));
-        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> BitsProxy.putInt(BUFF, BUFF.length, 1));
+        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> ByteArray.getInt(BUFF, BUFF.length));
+        assertThrowsOriginal(IndexOutOfBoundsException.class, () -> ByteArray.setInt(BUFF, BUFF.length, 1));
+    }
+
+    // At-zero methods
+
```

Comment:
These *AtZero* test methods don't add much value?  It is already testing an offset of 1 in the original tests.